logzio-sender/src/main/java/io/logz/sender/InMemoryLogsBuffer.java

```diff
@@ -0,0 +1,101 @@
+package io.logz.sender;
+
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+
+import java.io.IOException;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class InMemoryLogsBuffer implements LogzioLogsBufferInterface {
+    private static final int ONE_MEGA_BYTE_IN_BYTES = 1024 * 1024;
+    public static int DONT_LIMIT_BUFFER_SPACE = -1;
+    private final ConcurrentLinkedQueue<byte[]> logsBuffer;
+    private final boolean dontCheckEnoughMemorySpace;
+    private final int bufferThreshold;
+    private final SenderStatusReporter reporter;
+    private AtomicInteger size;
+
+    private InMemoryLogsBuffer(boolean dontCheckEnoughMemorySpace, int bufferThreshold, SenderStatusReporter reporter)
+            throws LogzioParameterErrorException {
+
+        logsBuffer = new ConcurrentLinkedQueue<>();
+        this.dontCheckEnoughMemorySpace = dontCheckEnoughMemorySpace;
+        this.bufferThreshold = bufferThreshold;
+        this.reporter = reporter;
+        this.size = new AtomicInteger();
+    }
+
+    @Override
+    public void enqueue(byte[] log) {
+        if(isEnoughSpace()) {
+            logsBuffer.add(log);
```

Comment:
You have two lines which are a transaction. If you have 1500 threads logging all the same time, you can easily add and only then see you are over your capacity.
You should wrap with ReentrantLock (see javadoc) on accessing the queue (add/remove)